### ISSUE TITLE
Map TaxonicName and -Concept to CoL resource

### ISSUE THREAD

**Issue by retog** [plazi/gg2rdf#23]
The narrowest matching supertaxon should be referred to without duplicating the names. Where the hierarchical names in the treatment do not match a taxon in CoL new resources shall be created up to the matching hierarchy top.

Where there is no matching CoL-Taxon the newly created resources should be created with the same style and ontology, so that clients cn only distinguish CoL from TB Taxa by their URIs.

**Comment by nleanba** [comment]
Could you please elaborate or provide an example? I don't fully understand.

1. If a treatment concerns a taxon-concept which has a matching taxon in CoL:
   - should the treatment directly aument/deprecate/define the CoL-taxon?
   - should info provided in the treatment (e.g. authority) be ignored or added to the CoL-taxon?
2. What happens to the distinction between TaxonNames and TaxonConcepts? The CoL resources (as generated by https://github.com/plazi/catologueoflife-to-rdf dont have a rdf type, but seem to match TaxonConcept (authority is provided)
3. The ontology of the CoL-resources doesn't map 1:1 onto the currently used ontology. How should names be handled that don't have an obvoius translation?
    e.g. what happens to "undef-species" in https://github.com/plazi/treatments-xml/blob/035dcb37eae7c6acb3782e4a71d6d0b5cede1490/data/39/8F/A8/398FA8D86705EFEA3B756FB9097A7E9D.xml#L92 ?

**Comment by retog** [comment]
> 1. If a treatment concerns a taxon-concept which has a matching taxon in CoL:
>    
>    * should the treatment directly aument/deprecate/define the CoL-taxon?

Yes, wherever we currently have a TaxonConcept this should be replaced 

>    * should info provided in the treatment (e.g. authority) be ignored or added to the CoL-taxon?

Whenever the full name used in the treatment doesn't match a taxon in CoL a new Taxon shall be  created. The new Taxon shall use the same properties as the one from CoL and point to CoL-Parent, or if the parent taxon is also not in CoL recursively to a new Taxon.

> 2. What happens to the distinction between TaxonNames and TaxonConcepts? The CoL resources (as generated by https://github.com/plazi/catologueoflife-to-rdf dont have a rdf type, but seem to match TaxonConcept (authority is provided)

TaxonNames are basically used when no authority is available. I think the name without authority unambiguosly references a taxon in CoL so this should be used.

> 3. The ontology of the CoL-resources doesn't map 1:1 onto the currently used ontology. How should names be handled that don't have an obvoius translation?
>    e.g. what happens to "undef-species" in https://github.com/plazi/treatments-xml/blob/035dcb37eae7c6acb3782e4a71d6d0b5cede1490/data/39/8F/A8/398FA8D86705EFEA3B756FB9097A7E9D.xml#L92 ?

The respective RDF looks quite broken as it stands: https://github.com/plazi/treatments-rdf/blob/0063581b502aa0e5fdb6b11018b55cc391fe635f/data/39/8F/A8/398FA8D86705EFEA3B756FB9097A7E9D.ttl#L31-L31C89
neither is "cf" an accepted name, nor "undef-species" a rank. In CoL "Lepralielloidea" is a superfamily in the suborder "Flustrina" encompassing 14 families, 78 genera and 665 species

So, if a mentioned taxon name can be expressed as a hierarchy of taxa should be matched to CoL where this matches the names and only if the taxon in the treatment has an authority this must match as well.
